pkg/odo/cli/component/create.go

```diff
@@ -382,41 +382,40 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 			defaultComponentNamespace = client.Namespace
 		}
 
-		var catalogDevfileList catalog.DevfileComponentTypeList
 		var componentType string
 		var componentName string
 		var componentNamespace string
+		isDevfileRegistryPresent := true // defaulted to true since odo ships with a default registry set
+
+		// Component type: We provide devfile component list to let user choose
```

Comment:
We can't make the function `catalog.ListDevfileComponents` global because we have the case where creating devfile component from existing devfile, that case doesn't need this function so that make this function global can reduce the performance. This function only use by creating devfile component from registry (either interactive or direct mode)

Reply:
updated